### PR TITLE
fix: Fix Android scroll_to and scroll_to_exact

### PR DESCRIPTION
**Pull request by emen**
I need to use this method for some of my tests.

Fixed Android `scroll_to` and `scroll_to_exact` methods and their tests `rake android['android/element/generic.rb']`

### CODE REVIEW

**Comment by KazuCocoa** [comment]
@emen 
Thanks!

Could you explain me more?
For instance, current implementations don't work for `uiautomator2` or ...

**Comment by emen** [comment]
Hi @KazuCocoa,

I ran into a couple of issues with the current implementation.

1. `rake android['android/element/generic.rb']` fails with `scroll_to` and `scroll_to_exact`
2. Use `scroll_to_exact` as an example, given an element `text("Test")` on the screen, `scroll_to("Test")` scrolls to the element first, and then tries to find and scroll to an element with description "Test", and then an element with resource id `""`. 

Thanks!
Emen

File: lib/appium_lib/android/element/generic.rb [hunk 1]
```diff
@@ -39,12 +39,16 @@ def scroll_uiselector(content, index = 0)
     # @return [Element] the element scrolled to
     def scroll_to(text, scrollable_index = 0)
       text = %("#{text}")
-
-      args = scroll_uiselector("new UiSelector().textContains(#{text})", scrollable_index) +
-             scroll_uiselector("new UiSelector().descriptionContains(#{text})", scrollable_index) +
-             scroll_uiselector(resource_id(text, "new UiSelector().resourceId(#{text});"), scrollable_index)
-
-      find_element :uiautomator, args
+      rid  = resource_id(text, "new UiSelector().resourceId(#{text});")
```
Comment (KazuCocoa) [inline lib/appium_lib/android/element/generic.rb]: How about the following if you'd like to ignore ` resource_id(text, "new UiSelector().resourceId(#{text});")` is `""` case?

```ruby
r_id = resource_id(text, "new UiSelector().resourceId(#{text});")
args = scroll_uiselector("new UiSelector().textContains(#{text})", scrollable_index) +
        scroll_uiselector("new UiSelector().descriptionContains(#{text})", scrollable_index)
args += scroll_uiselector(r_id, scrollable_index) unless r_id.empty?

find_element :uiautomator, args
```

File: lib/appium_lib/android/element/generic.rb [hunk 2]
```diff
@@ -53,12 +57,16 @@ def scroll_to(text, scrollable_index = 0)
     # @return [Element] the element scrolled to
     def scroll_to_exact(text, scrollable_index = 0)
       text = %("#{text}")
-
-      args = scroll_uiselector("new UiSelector().text(#{text})", scrollable_index) +
-             scroll_uiselector("new UiSelector().description(#{text})", scrollable_index) +
-             scroll_uiselector(resource_id(text, "new UiSelector().resourceId(#{text});"), scrollable_index)
-
-      find_element :uiautomator, args
+      rid  = resource_id(text, "new UiSelector().resourceId(#{text});")
```
Comment (KazuCocoa) [inline lib/appium_lib/android/element/generic.rb]: Also,

```ruby
      r_id = resource_id(text, "new UiSelector().resourceId(#{text});")
      args = scroll_uiselector("new UiSelector().text(#{text})", scrollable_index) +
             scroll_uiselector("new UiSelector().description(#{text})", scrollable_index)
      args += scroll_uiselector(r_id, scrollable_index) unless r_id.empty?
```

**Comment by KazuCocoa** [comment]
@emen 
Thanks for your quick reply! I posted some comments.

**Comment by emen** [comment]
Hi @KazuCocoa,

I figured it out. The tests fail with Android 5.0.0. and 5.1.0, pass with 6.0.0 and 7.0.0. With 5.0.0 and 5.1.1, Ruby needs to check if the scroll is successful and return the element. 

Please try and let me know.

Thanks a lot!
Emen

**Comment by KazuCocoa** [comment]
@emen 
Hm... I tried with Android 5.0 and 5.1 on emulators, but they work fine.
Could you check the following lines in your environment as I've reviewed? And could you attach the results, Appium side and Ruby side, as Gist?

```ruby
r_id = resource_id(text, "new UiSelector().resourceId(#{text});")
args = scroll_uiselector("new UiSelector().textContains(#{text})", scrollable_index) +
        scroll_uiselector("new UiSelector().descriptionContains(#{text})", scrollable_index)
args += scroll_uiselector(r_id, scrollable_index) unless r_id.empty?

find_element :uiautomator, args
```

I'd like to reduce the number of request to the server, and I'd like to call `find_element` once as possible.

**Comment by emen** [comment]
Thank you @KazuCocoa!

You are right. With the above change, the unit test `rake android['android/element/generic.rb']`will pass. But that is because in the test app, the text and the description have the same content. It still failed with my app. Here is [Appium log](https://gist.github.com/emen/2afb399e573807f053514f7db5b37338). The element have text 'Current Environment', but description is something different. Please let me know if you need more details

Thanks!
Emen

**Comment by KazuCocoa** [comment]
thanks!

**Pull request merged**